File: 1874 - 스택 수열/1874.py
import sys


class Stack:
    def __init__(self):
        self.items = []
        self.size = 0

    def isEmpty(self):
        return self.items == []

    def push(self, item):
        self.items.append(item)
        self.size += 1

    def pop(self):
        self.size -= 1
        return self.items.pop()

    def size(self):
        return self.size

    def top(self):
        return self.items[-1]


read = sys.stdin.readline
t = int(read())
s = Stack()
s.push(0)
result = []
flag = 0
item = 1
for _ in range(t):
    n = int(read())
    top = s.top()
    if n >= item:
        for i in range(n-item+1):
            s.push(item)
            item += 1
            result.append('+')
    elif top != n:
        flag = 1
    _ = s.pop()
    result.append('-')
if flag == 1:
    print("NO")
else:
    for i in range(len(result)):
        print(result[i])

File: 1874 - 스택 수열/1874.cs
using System;
using System.Text;
using System.Collections.Generic;


namespace _1874
{
    class Program
    {
        static void Main(string[] args)
        {
            int t = int.Parse(Console.ReadLine());
            Stack<int> s = new Stack<int>();
            StringBuilder sb = new StringBuilder();
            s.Push(0);
            int n;
            int item = 1;
            bool flag = false;
            for (int _ = 0; _ < t; _++)
            {
                n = int.Parse(Console.ReadLine());
                if (n >= item)
                {
                    for (; item <= n; item++) {
                        s.Push(item);
                        sb.AppendLine("+");
                    }
                }
                else if (n != s.Peek())
                {
                    flag = true;
                    break;
                }
                s.Pop();
                sb.AppendLine("-");
            }
            Console.WriteLine(flag ? "NO" : sb.ToString());
        }
    }
}

